osu.Game/Online/API/Requests/SearchBeatmapSetsRequest.cs

```diff
@@ -35,6 +37,9 @@ protected override WebRequest CreateWebRequest()
             if (ruleset.ID.HasValue)
                 req.AddParameter("m", ruleset.ID.Value.ToString());
 
+            if (cursor != null)
+                req.AddParameter("cursor", cursor.ToString());
```

Comment:
I doubt that this is the right way to actually add the cursor to the request. The result is that the pager just receives the same result every time. What's the proper way to do this?

Reply:
The problem here is that web actually expects _two_ parameters, ie. in the form of
```
GET /search?cursor[approved_date]=1586609019000&cursor[_id]=1112051
```
which requires lazer to be able to reinterpret that JSON structure somehow to split out the fields and send them back to web in the format required.

Reply:
Are `approved_date` and `_id` always the only two cursor parameters, or does it vary based on the sort...?

Reply:
[Seems like they aren't](https://github.com/ppy/osu-web/blob/4687a1e8a5c132af3ccfc63a2b90d239ec395d71/app/Libraries/Search/BeatmapsetSearchRequestParams.php#L153-L212), which makes me think a more robust solution is needed (maybe a way to automatically unwrap all fields of `cursor` into the naming convention web expects?)

Reply:
Perhaps the cursor can be represented using a `Dictionary<string, string>`?

Reply:
I mean, that seems like it would *work*, but it assumes there are no further nested objects (which I can't guarantee right now) and also isn't the prettiest solution. In my mind the absolute cleanest way to handle this would be to not handle it and leave it up to web.

In general I'd actually review all existing requests that use cursors - maybe they've been broken this whole time but nobody actually noticed, since it seems like a long shot this popped up only now.

Reply:
Oh, and if this re-mapping required currently does stay lazer-side, ideally there should be a method like
```csharp
req.AddCursor(cursor);
```
that does the mapping automagically internally.

Reply:
I think that regardless of how it's done, some level of parsing will be required due to the way the parameters have to be added. Do you have any suggestions for how you would like to see this handled? On the topic of other requests that use cursors, I think that pagination via cursors simply isn't implemented anywhere else, so nothing is broken, just unimplemented.

Reply:
If web accepted the same JSON object it sent out, it would require no interpretation on the client's side, which seems better to me (there's no reason for clients to parse the state of the cursor, it's just returned by API only to prevent the necessity of API having to keep track of client state for paging), which is why I'm partially reluctant of suggesting any particular way of handling this client-side.

Then again, putting a JSON blob in a query string is not exactly pretty, either; can't put it in a `GET` body either because that breaks HTTP spec.

I'd actually like the team's opinion on this as I see no one best path forward.